flow/scripts/cts.tcl

```diff
@@ -41,7 +41,10 @@ if {[info exist ::env(CTS_SNAPSHOTS)]} {
   save_progress 4_1_pre_repair_clock_nets
 }
 
-set_propagated_clock [all_clocks]
+
+if {![info exist ::env(SDC_IDEAL_CLOCK)]} {
+    set_propagated_clock [all_clocks]
+}
```

Comment:
I have the same concern about inside the block reg-to-reg paths not seeing any skew.

Reply:
This skew can be covered by the clock uncertainty which is in the SDC. Since the element is small this should also be a small predictable value. I was on the fence about adding the propagated clocks back into the final report. This would detect if the clock uncertainty was too low and there were still reg to reg failures.

It really a preference. The approach in this PR will work. 

Another option is the one Matt proposed which is to leave everything alone and false path the i/o paths when the element is being constructed.  set_false_path -hold could be used so that only hold timing from the top would be ignored. Setup would still be check and optimized for the element top level paths.

Reply:
The approach demonstrated should hold for a large Element. mock-array should have the minimum complication required to illustrate solutions for more complicated real world problems. A use case is BoomTile from MegaBoom...